.github/workflows/docs-build-push.yml

```diff
@@ -18,12 +18,29 @@ on:
     paths:
     - "site/**"
 
+concurrency:
+  group: ${{ github.ref_name }}-docs-push
+  cancel-in-progress: true
+
 permissions:
   contents: read
 
 jobs:
+  vars:
+    runs-on: ubuntu-22.04
+    outputs:
+      azure_creds: ${{ steps.vars.outputs.defined }}
+    steps:
+    - name: Check if variable is set
+      id: vars
+      env:
+        AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS_DOCS }}
+      if: "${{ env.AZURE_CREDENTIALS != '' }}"
+      run: echo "defined=true" >> $GITHUB_OUTPUT
+
   call-docs-build-push:
-    if: ${{ github.event.repository.fork == false }}
+    needs: [vars]
+    if: ${{ github.event.repository.fork == false && needs.vars.outputs.azure_creds == 'true' }}
```

Comment:
isn't it just enough to check if it's a fork?

Reply:
Apparently not because the `pull_request` trigger still ran it.